src/aspire/utils/coor_trans.py

```diff
@@ -302,3 +304,29 @@ def common_line_from_rots(r1, r2, ell):
     ell_ji = int(np.mod(np.round(ell_ji), ell))
 
     return ell_ij, ell_ji
+
+
+def crop_2d(mat, size, fill_value=0):
+    """
+    :param mat: A 2-dimensional numpy array
+    :param size: Integer size of cropped/padded output
+    :return: A numpy array of shape (size, size)
+    """
+
+    mat_x, mat_y = mat.shape
+    # shift terms
+    start_x = math.floor(mat_x / 2) - math.floor(size / 2)
+    start_y = math.floor(mat_y / 2) - math.floor(size / 2)
+
+    # cropping
+    if size <= min(mat_x, mat_y):
+        return mat[start_x : start_x + size, start_y : start_y + size]
+    # padding
+    elif size >= max(mat_x, mat_y):
+        # ensure that we return in the same dtype as the input
+        to_return = fill_value * np.ones((size, size), dtype=mat.dtype)
+        to_return[-start_x : mat_x - start_x, -start_y : mat_y - start_y] = mat
```

Comment:
I'm confused. Why the negative indices?

Reply:
This is from the original implementation. `start_x` and `start_y` are actually themselves negative when cropping because `size` is larger than `mat_x` and `mat_y`.

```python
# shift terms                                                                                                          
    start_x = math.floor(im_x / 2) - math.floor(size / 2)
    start_y = math.floor(im_y / 2) - math.floor(size / 2)
```

Reply:
It basically calculates the starting point of where we put the original image, relative to the new (larger) box / frame around it.

Reply:
Huh. May be worth a clarifying comment.